src/hotspot/cpu/x86/stubGenerator_x86_64_arraycopy.cpp

```diff
@@ -2476,6 +2478,286 @@ address StubGenerator::generate_unsafe_copy(const char *name,
 }
 
 
+//  Generate 'unsafe' set memory stub
+//  Though just as safe as the other stubs, it takes an unscaled
+//  size_t argument instead of an element count.
+//
+//  Input:
+//    c_rarg0   - destination array address
+//    c_rarg1   - byte count (size_t)
+//    c_rarg2   - byte value
+//
+// Examines the alignment of the operands and dispatches
+// to an int, short, or byte fill loop.
+//
+address StubGenerator::generate_unsafe_setmemory(const char *name) {
+  __ align(CodeEntryAlignment);
+  StubCodeMark mark(this, "StubRoutines", name);
+  address start = __ pc();
+
+  // bump this on entry, not on exit:
+  INC_COUNTER_NP(SharedRuntime::_unsafe_set_memory_ctr, rscratch1);
+
+  {
+    Label L_exit, L_fillQuadwords, L_fillDwords, L_fillBytes;
+
+    setup_arg_regs(3);
+
+    const Register dest = rdi;
+    const Register size = rsi;
+    const Register wide_value = rax;
+    const Register rScratch1 = rcx;
+    const Register rScratch3 = r8;
+    const Register rScratch4 = r11;
+
+    //     fill_to_memory_atomic(unsigned char*, unsigned long, unsigned char)
+
+    __ testq(size, size);
+    __ jcc(Assembler::zero, L_exit);
+
+    {
+      const Register byteVal = rdx;
+
+      // Propagate byte to full Register
```

Comment:
The comment refers to lines 2524-2526, please move it down.

Reply:
Still continuing to look through StubGenerator::generate_unsafe_setmemory(), more comments to come. Thank you for your patience.

Reply:
Comment moved.